substrate/bin/node/runtime/src/genesis_config_presets.rs

```diff
@@ -0,0 +1,227 @@
+use sp_core::{crypto::get_public_from_string_or_panic, sr25519};
+use sp_genesis_builder::PresetId;
+use sp_keyring::Sr25519Keyring;
+use sp_mixnet::types::AuthorityId as MixnetId;
+use sp_runtime::Perbill;
+
+pub const ENDOWMENT: Balance = 10_000_000 * DOLLARS;
+pub const STASH: Balance = ENDOWMENT / 1000;
+
+pub struct StakingPlaygroundConfig {
+	/// (Validators, Nominators)
+	pub dev_stakers: (u32, u32),
+	pub validator_count: u32,
+	pub minimum_validator_count: u32,
+}
+
+/// The staker type as supplied ot the Staking config.
+pub type Staker = (AccountId, AccountId, Balance, StakerStatus<AccountId>);
+
+/// Helper function to create RuntimeGenesisConfig json patch for testing.
+pub fn kitchen_sink_genesis(
+	initial_authorities: Vec<(AccountId, AccountId, SessionKeys)>,
+	root_key: AccountId,
+	endowed_accounts: Vec<AccountId>,
+	stakers: Vec<Staker>,
+	staking_playground_config: Option<StakingPlaygroundConfig>,
+) -> serde_json::Value {
+	let (validator_count, min_validator_count, dev_stakers) = match staking_playground_config {
+		Some(c) => (c.validator_count, c.minimum_validator_count, Some(c.dev_stakers)),
+		None => {
+			let authorities_count = initial_authorities.len() as u32;
+			(authorities_count, authorities_count, None)
+		},
+	};
+
+	let collective = collective(&endowed_accounts);
+
+	build_struct_json_patch!(RuntimeGenesisConfig {
+		balances: BalancesConfig {
+			balances: endowed_accounts.iter().cloned().map(|x| (x, ENDOWMENT)).collect(),
+			..Default::default()
+		},
+		session: SessionConfig {
+			keys: initial_authorities
+				.iter()
+				.map(|x| { (x.0.clone(), x.1.clone(), x.2.clone()) })
+				.collect(),
+		},
+		staking: StakingConfig {
+			validator_count,
+			minimum_validator_count: min_validator_count,
+			invulnerables: initial_authorities
+				.iter()
+				.map(|x| x.0.clone())
+				.collect::<Vec<_>>()
+				.try_into()
+				.expect("too many authorities"),
+			slash_reward_fraction: Perbill::from_percent(10),
+			stakers,
+			dev_stakers
+		},
+		elections: ElectionsConfig {
+			members: collective.iter().cloned().map(|member| (member, STASH)).collect(),
+		},
+		technical_committee: TechnicalCommitteeConfig { members: collective },
+		sudo: SudoConfig { key: Some(root_key) },
+		babe: BabeConfig { epoch_config: BABE_GENESIS_EPOCH_CONFIG },
+		society: SocietyConfig { pot: 0 },
+		assets: AssetsConfig {
+			// This asset is used by the NIS pallet as counterpart currency.
+			assets: vec![(9, Sr25519Keyring::Alice.to_account_id(), true, 1)],
+			..Default::default()
+		},
+		nomination_pools: NominationPoolsConfig {
+			min_create_bond: 10 * DOLLARS,
+			min_join_bond: 1 * DOLLARS,
+		},
+	})
+}
+
+/// Provides the JSON representation of predefined genesis config for given `id`.
+pub fn get_preset(id: &PresetId) -> Option<Vec<u8>> {
+	// Note: Can't use `Sr25519Keyring::Alice.to_seed()` because the seed comes with `//`.
+	let (alice_stash, alice, alice_session_keys) = authority_keys_from_seed("Alice");
```

Comment:
This was the problem why the node did not produce blocks. I generated the wrong accounts with `authority_keys_from_seed(Sr25519Keyring::Alice.to_seed())` 🙈